packages/components/app/styles/components/accordion.scss

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// ACCORDION
+//
+
+@use "../mixins/button" as *;
+@use "../mixins/focus-ring" as *;
+
+:root {
+  --token-accordion-item-padding: 16px;
+}
+
+// ACCORDION COMPONENT (wrapper)
+
+.hds-accordion {
+  display: flex;
+  flex-direction: column;
+  gap: 12px;
+
+  // override default .hds-disclosure-primitive style (Q: Should primitive be refactored?)
+  .hds-accordion-item { width: 100%; }
+}
+
+// ACCORDION ITEM COMPONENT (nested child)
+
+.hds-accordion-item {
+  border-radius: 6px;
+
+  // isClickable=true, Entire AccordionItem is clickable:
+  &.hds-accordion-item--is-clickable {
+    box-shadow: var(--token-surface-mid-box-shadow);
+
+    &:hover,
+    &.mock-hover {
+      box-shadow: var(--token-surface-high-box-shadow);
+    }
+
+    &.mock-focus {
+      > .hds-disclosure-primitive__toggle .hds-accordion-item__toggle {
+        @include hds-focus-ring-with-pseudo-element();
+
+        &::before {
+          box-shadow: var(--token-focus-ring-action-box-shadow);
+        }
+      }
+    }
```

Comment:
I know the focus of this review is meant to be on the API, but this should not be needed. You can use attributes to specify on which item you want the 'mocking' to happen using `mock-state-value` and `mock-state-selector` (see [tag.hbs](https://github.com/hashicorp/design-system/blob/main/packages/components/tests/dummy/app/templates/components/tag.hbs#L39)).

It currently resolves in 'double' focus
<img width="515" alt="Screenshot 2023-06-26 at 19 01 45" src="https://github.com/hashicorp/design-system/assets/788096/6fd950f7-20bd-4494-98f9-d37701d0a173">

Reply:
Thanks. I wasn't sure how to do that. I'll take a look at the example you linked.

Reply:
Pushed up some fixes.

Reply:
Sorry I was confusing myself about where the focus was coming from. I pushed up another fix that should be correct now.

Reply:
Unresolving...

The changes I pushed up previously messed up the box-shadow position so it was only around the toggle and not around the content too so I undid that change.

So I still have the problem that the box shadow makes the AccordionItem look a bit taller since it's outside the containing element (also due to the transparent border I added to the match the gray border around the other variant). The mock-hover state is also no longer working in the Showcase. So I need to try to figure out how to fix those items...

Reply:
Ok this should now finally be resolved. Part of my issue earlier was I didn't realize the designs for the 2 versions (`@isInteractive=true/false`) had been updated to both use a box-shadow vs. one having a border instead.